src/package/components/button/Button.js

```diff
@@ -0,0 +1,23 @@
+import React from 'react'
+
+const preventDefault = (callback) => (e) => {
+  e.preventDefault();
+  if(callback) {
+    callback(e);
+  }
+};
+
+export default ({ children, id, onClick, className, style, type }) => {
+  const buttonType = type || 'button'; // Make button default so that when used in forms, "submit" does not become default
```

Comment:
Instead of creating a variable inside the component, it may would be better to define the expected default value directly to the component props like:

```jsx
const Button = ({ ...otherProps, type }) => (
  // 
  // Button component implementation
  // 
)

// here we define the expected default value to the `type` prop
Button.defaultProps = { 
  type: 'button'
}

export default Button
```